dev_guide/builds.adoc

```diff
@@ -11,11 +11,79 @@ toc::[]
 
 == Overview
 A link:../architecture/core_objects/builds.html[build] is a process of creating
-runnable images to be used on OpenShift. There are three types of builds:
+runnable images to be used on OpenShift. There are three build strategies:
 
-- Docker build
-- STI build
-- custom build
+- link:../architecture/core_objects/builds.html#source-build[Source-To-Image]
+- link:../architecture/core_objects/builds.html#docker-build[Docker]
+- link:../architecture/core_objects/builds.html#custom-build[Custom]
+
+== Defining a buildConfig
+
+A `*buildConfig*` describes a single build definition and a
+set of link:#triggers[triggers] for when a new build should be created.
+
+A buildConfig is a REST object, which can be used in a POST to the API server to
+create a new instance. The following configuration results in a new build
+every time a Docker image tag or the source code changes:
+
+[source.json]
+----
+{
+  "kind": "BuildConfig",
+    "apiVersion": "v1beta3",
+    "metadata": {
+      "name": "ruby-sample-build <1>",
+    },
+  "spec": {
+    "triggers": [ <2>
+      {
+        "type": "github",
+        "github": {
+          "secret": "secret101"
+        }
+      },
+      {
+        "type": "generic",
+        "generic": {
+          "secret": "secret101"
+        }
+      },
+      {
+        "type": "ImageChange"
+      }
+    ],
+    "source": { <3>
+      "type": "Git",
+      "git": {
+        "uri": "git://github.com/openshift/ruby-hello-world.git"
+      }
+    },
+    "strategy": { <4>
+      "type": "Source",
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "ruby-20-centos7:latest"
+        },
+        "incremental": true
```

Comment:
Don't do incremental here, I'm writing a small paragraph about it, as a follow-up PR to yours.